### DISCUSSION TITLE
GainPowerBackonHypers constant

### DISCUSSION THREAD

**Discussion by jenngra505** [ikemen-engine/Ikemen-GO#2068] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

Many beginner creators in Mugen would forget to add getpower = 0 to their super moves resulting in a glaring oversight.

### Describe the solution you'd like

GainPowerBackonHypers is a boolean you can add to the Data sections of the constants file to determine if this happens. It would detect the attribution of an attack, and if it's a hyper, it won't give power back when set to 0. Mugen characters will default to 1, while Ikemenversion characters will default to 0. I haven't asked for this as a hardcoded thing for Ikemenversion in case there's an obscure kusoge game where the attacker gains power back when using a hyper but also does things that can only be implemented in Ikemen.

**Comment by potsmugen** [comment, marked as answer]
While I don't think the engine should babysit char stuff too much, a `Super.Attack.LifeToPowerMul` constant would be perfectly in line with the rest of the code.
